scripts/prepare_dutchf3.py

```diff
@@ -105,66 +128,118 @@ def split_patch_train_val(data_dir, stride, patch_size, per_val=0.2, log_config=
+                         for k in vert_locations)
             for j, k in locations:
                 yield "i_" + str(i) + "_" + str(j) + "_" + str(k)
 
-    test_iline_range = list(test_iline_range)
-    test_i_list = list(_i_extract_patches(test_iline_range, horz_locations, vert_locations))
-    train_i_list = list(_i_extract_patches(train_iline_range, horz_locations, vert_locations))
+    # Process inlines - train
+    horz_locations_train = range(val_iline, xline - patch_size, stride)
+    logger.debug("Generating Inline patches")
+    logger.debug("Generating Inline patches - Train")
+    logger.debug(horz_locations_train)
+    train_i_list = list(_i_extract_patches(train_iline_range,
+                                          horz_locations_train,
+                                          vert_locations))
+    # val_iline - define size of the validation set for the fist part
+    val_iline_range = list(val_iline_range)
+
+    # Process inlines - validation - begining
+    logger.debug("Generating Inline patches - validation")
+    horz_locations_val_begin = range(0, val_iline, max(1,stride))
+    val_i_list = list(_i_extract_patches(val_iline_range,
+                                           horz_locations_val_begin,
+                                           vert_locations))
+    # Process inlines - validation - end
+    horz_locations_val_end = range(xline - stride + 1, xline, max(1,stride))
+    val_i_list += list(_i_extract_patches(val_iline_range,
+                                                horz_locations_val_end,
+                                                vert_locations))
+
     logger.debug(train_iline_range)
-    logger.debug(test_iline_range)
+    logger.debug(val_iline_range)
 
     # Process crosslines
-    horz_locations = range(0, iline - patch_size + 1, stride)
-    vert_locations = range(0, depth - patch_size + 1, stride)
-    logger.debug("Generating Crossline patches")
-    logger.debug(horz_locations)
-    logger.debug(vert_locations)
-
     def _x_extract_patches(xline_range, horz_locations, vert_locations):
         for j in xline_range:
-            locations = ([i, k] for i in horz_locations for k in vert_locations)
+            locations = ([i, k] for i in horz_locations
+                         for k in vert_locations)
             for i, k in locations:
                 yield "x_" + str(i) + "_" + str(j) + "_" + str(k)
-    test_xline_range = list(test_xline_range)
-    test_x_list = list(_x_extract_patches(test_xline_range, horz_locations, vert_locations))
-    train_x_list = list(_x_extract_patches(train_xline_range, horz_locations, vert_locations))
+
+    logger.debug("Generating Crossline patches")
+    logger.debug("Generating Crossline patches - Train")
+    # xline = iline x depth
+    val_xline = math.floor(iline * per_val / 2)
+    logger.debug(val_xline)
+
+    # Process xlines - train
+    horz_locations_train = range(val_xline, iline - patch_size, stride)
```

Comment:
not sure i'm following here. Why is the range from val_xline to iline - patch_size? <-- it might be easier to jump on a quick call to discuss this.